realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SubscriptionSetTests.kt

```diff
@@ -0,0 +1,291 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.mongodb.sync
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import io.realm.Realm
+import io.realm.TEST_APP_3
+import io.realm.TestApp
+import io.realm.TestHelper
+import io.realm.TestSyncConfigurationFactory
+import io.realm.entities.SyncColor
+import io.realm.kotlin.where
+import io.realm.mongodb.close
+import io.realm.mongodb.registerUserAndLogin
+import io.realm.rule.BlockingLooperThread
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.TimeUnit
+import kotlin.test.assertFailsWith
+
+/**
+ * Class wrapping tests for SubscriptionSets
+ */
+@RunWith(AndroidJUnit4::class)
+class SubscriptionSetTests {
+
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+    private val looperThread = BlockingLooperThread()
+
+    private lateinit var app: TestApp
+    private lateinit var realmConfig: SyncConfiguration
+    private lateinit var realm: Realm
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        app = TestApp(appName = TEST_APP_3)
+        val user = app.registerUserAndLogin(TestHelper.getRandomEmail(), "123456")
+        realmConfig = configFactory.createFlexibleSyncConfiguationBuilder(user)
+            .schema(SyncColor::class.java)
+            .syncClientResetStrategy { session, error -> fail("Client Reset should not trigger.") }
+            .build()
+        realm = Realm.getInstance(realmConfig)
+    }
+
+    @After
+    fun tearDown() {
+        realm.close()
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    @Test
+    fun initialSubscriptions() {
+        val subscriptions = realm.subscriptions
+        assertEquals(0, subscriptions.size())
+        assertEquals(SubscriptionSet.State.UNCOMMITTED, subscriptions.state)
+    }
+
+    @Test
+    fun find() {
+        val subscriptions = realm.subscriptions
+        assertNull(subscriptions.find(realm.where<SyncColor>()))
+        subscriptions.update { mutableSubs ->
+            mutableSubs.addOrUpdate(Subscription.create(realm.where<SyncColor>()))
+        }
+        assertNotNull(subscriptions.find(realm.where<SyncColor>()))
```

Comment:
Should we just for completion check that the actual type of the returned subscription is right ;)